vms/proposervm/block/build.go

```diff
@@ -47,34 +160,37 @@ func Build(
 	blockBytes []byte,
 	chainID ids.ID,
 	key crypto.Signer,
+	blockVrfSig []byte,
 ) (SignedBlock, error) {
-	block := &statelessBlock{
+	block := &statelessBlock[statelessUnsignedBlock]{
 		StatelessBlock: statelessUnsignedBlock{
 			ParentID:     parentID,
 			Timestamp:    timestamp.Unix(),
 			PChainHeight: pChainHeight,
 			Certificate:  cert.Raw,
 			Block:        blockBytes,
+			VRFSig:       blockVrfSig,
 		},
 		timestamp: timestamp,
 		cert:      cert,
 		proposer:  ids.NodeIDFromCert(cert),
 	}
-	var blockIntf SignedBlock = block
 
-	unsignedBytesWithEmptySignature, err := Codec.Marshal(CodecVersion, &blockIntf)
-	if err != nil {
+	var err error
+
+	var blkBytes []byte
```

Comment:
nit: is there a reason we declare these here instead of with `:=` like this?

```go
blkBytes, err := marshalBlock(block)
```

Reply:
done